rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -258,6 +260,53 @@ where
     }
 }
 
+pub(crate) struct AuthorizeContext<C> {
+    credential: C,
+    allow_anonymous: bool,
+}
+
+impl<C> AuthorizeContext<C> {
+    pub(crate) fn new(credential: C) -> Self {
+        Self {
+            credential,
+            allow_anonymous: false,
+        }
+    }
+}
+
+impl<S, B, C: ValidateRequest<S, B>> ValidateRequest<S, B> for AuthorizeContext<C>
+where
+    S: Clone + Send + Sync + 'static,
+    B: Send + 'static,
+{
+    type ResponseBody = C::ResponseBody;
+
+    async fn validate(
+        &self,
+        ctx: Context<S>,
+        req: Request<B>,
+    ) -> Result<(Context<S>, Request<B>), Response<Self::ResponseBody>> {
+        match req.headers().get(header::AUTHORIZATION) {
+            Some(_) => self.credential.validate(ctx, req).await,
+            None if self.allow_anonymous => {
+                ctx.insert(UserId::anonymous());
+
+                Ok((ctx, req))
+            }
+            None => self.credential.validate(ctx, req).await,
+        }
+    }
+}
+
+impl<C: Clone> Clone for AuthorizeContext<C> {
```

Comment:
also implement `fmt::Debug` like this please

Reply:
added 922cfc4eaf06ec4ad9964cafb71520a3aa8e02f3